### PR TITLE
Swerve Superstructure -> states-dev

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -19,16 +25,23 @@ public class Robot {
...
+
 	public Robot() {
 		this.superstructure = new Superstructure(this);
 		configureBindings();
+
+		this.swerve = new Swerve(
+				SwerveConstantsFactory.create(SwerveType.SWERVE),
+				ModulesFactory.create(SwerveType.SWERVE),
+				GyroFactory.create(SwerveType.SWERVE)
+		);
```
Comment (YHerm) [inline src/main/java/frc/robot/Robot.java]: first thing (put above)

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -19,16 +25,23 @@ public class Robot {
 
 	private final Superstructure superstructure;
 
+	private final Swerve swerve;
```
Comment (YHerm) [inline src/main/java/frc/robot/Robot.java:25]: above!

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -37,4 +50,7 @@ public Superstructure getSuperstructure() {
 		return superstructure;
 	}
 
+	public Swerve getSwerve() {
+		return swerve;
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/Robot.java]: above superstructure

File: src/main/java/frc/robot/Robot.java
Comment (YHerm) [inline src/main/java/frc/robot/Robot.java:1]: add in joystickbindings first joystick buttons for swerve
(reset heading and drive)

Comment (drorta) [reply src/main/java/frc/robot/Robot.java:1]: wdym drive

**Review by Nitay4 — DISMISSED** [review]

**Review by YHerm — DISMISSED** [review]
Make sure tot test it

**Review by Yotam7 — DISMISSED** [review]

File: src/main/java/frc/robot/constants/IDs.java [hunk 4]
```diff
@@ -11,4 +13,38 @@ public class IDs {
 		PowerDistribution.ModuleType.kRev
 	);
 
+	public static final Phoenix6DeviceID PIGEON_2_DEVICE_ID = new Phoenix6DeviceID(0, BusChain.CANIVORE);
+
+	public static class TalonFXIDs {
+
+		public static final Phoenix6DeviceID FRONT_LEFT_STEER_MOTOR = new Phoenix6DeviceID(0, BusChain.CANIVORE);
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/constants/IDs.java]: No need for "MOTOR" in any of them

File: src/main/java/frc/robot/constants/IDs.java [hunk 5]
```diff
@@ -11,4 +13,38 @@ public class IDs {
 		PowerDistribution.ModuleType.kRev
 	);
 
+	public static final Phoenix6DeviceID PIGEON_2_DEVICE_ID = new Phoenix6DeviceID(0, BusChain.CANIVORE);
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/constants/IDs.java]: No need for "DEVICE_ID" its clear

File: src/main/java/frc/robot/subsystems/swerve/factories/modules/constants/RealModuleConstants.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package frc.robot.subsystems.swerve.factories.modules.constants;
+
+import frc.robot.subsystems.swerve.SwerveType;
+import frc.robot.subsystems.swerve.factories.swerveconstants.RealSwerveConstants;
+import frc.robot.subsystems.swerve.module.ModuleConstants;
+import frc.robot.subsystems.swerve.module.ModuleUtils;
+
+public class RealModuleConstants {
+
+	private static final double WHEEL_DIAMETER_METERS = 0.048359 * 2;
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/swerve/factories/modules/constants/RealModuleConstants.java:10]: Space

File: src/main/java/frc/robot/subsystems/swerve/factories/swerveconstants/RealSwerveConstants.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package frc.robot.subsystems.swerve.factories.swerveconstants;
+
+import com.pathplanner.lib.util.PIDConstants;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveType;
+
+public class RealSwerveConstants {
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/swerve/factories/swerveconstants/RealSwerveConstants.java:9]: I think its nic3r with spaces and souble spaces

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Pull request merged**